crates/net/network/src/peers/manager.rs

```diff
@@ -656,6 +709,15 @@ impl Default for PeersManager {
     }
 }
 
+impl Drop for PeersManager {
+    fn drop(&mut self) {
+        let Some(file_path) = self.peers_file.take() else { return };
+        if let Err(err) = self.dump_peers(&file_path) {
+            warn!(target = "net::peers", ?err, file = ?file_path.display(), "Failed to save peers to file")
+        }
+    }
```

Comment:
this won't work because it won't fire on SIGNT (ctrl-c)

Reply:
crazy

Reply:
Really crazy 😵‍💫